presto-main/src/main/java/com/facebook/presto/metadata/InternalNodeManager.java

```diff
@@ -32,4 +33,8 @@
     AllNodes getAllNodes();
 
     void refreshNodes();
+
+    void addAllNodesListener(Consumer<AllNodes> listener);
```

Comment:
I think it might be clearer to call it something like `addNodeChangeListener`. The fact that it's an `AllNodes` seems less relevant, and makes the interface more confusing.